src/main/java/org/folio/services/iteration/IterationJobRunner.java

```diff
@@ -0,0 +1,201 @@
+package org.folio.services.iteration;
+
+import static io.vertx.core.Future.succeededFuture;
+import static org.folio.Environment.environmentName;
+import static org.folio.persist.InstanceRepository.INSTANCE_TABLE;
+import static org.folio.rest.jaxrs.model.IterationJob.JobStatus.IDS_PUBLISHED;
+import static org.folio.rest.jaxrs.model.IterationJob.JobStatus.ID_PUBLISHING_CANCELLED;
+import static org.folio.rest.jaxrs.model.IterationJob.JobStatus.ID_PUBLISHING_FAILED;
+import static org.folio.rest.jaxrs.model.IterationJob.JobStatus.PENDING_CANCEL;
+import static org.folio.rest.tools.utils.TenantTool.tenantId;
+
+import java.util.Map;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.WorkerExecutor;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowStream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.persist.IterationJobRepository;
+import org.folio.rest.jaxrs.model.Instance;
+import org.folio.rest.jaxrs.model.IterationJob;
+import org.folio.rest.persist.PgUtil;
+import org.folio.rest.persist.PostgresClientFuturized;
+import org.folio.rest.persist.SQLConnection;
+import org.folio.services.domainevent.CommonDomainEventPublisher;
+import org.folio.services.domainevent.DomainEvent;
+import org.folio.services.domainevent.DomainEventType;
+import org.folio.services.kafka.InventoryProducerRecordBuilder;
+import org.folio.services.kafka.topic.KafkaTopic;
+
+public class IterationJobRunner {
+
+  public static final String ITERATION_JOB_ID_HEADER = "iteration-job-id";
+
+  private static final Logger log = LogManager.getLogger(IterationJobRunner.class);
+  private static final int POOL_SIZE = 2;
+  private static final int UPDATE_PUBLISHED_EVERY = 1000;
+  private static volatile WorkerExecutor workerExecutor;
+
+  private final Context vertxContext;
+  private final Map<String, String> okapiHeaders;
+  private final PostgresClientFuturized postgresClient;
+  private final IterationJobRepository repository;
+  private CommonDomainEventPublisher<Instance> eventPublisher;
+
+
+  public IterationJobRunner(Context vertxContext, Map<String, String> okapiHeaders) {
+    this(new PostgresClientFuturized(PgUtil.postgresClient(vertxContext, okapiHeaders)),
+      new IterationJobRepository(vertxContext, okapiHeaders),
+      vertxContext,
+      okapiHeaders);
+  }
+
+  public IterationJobRunner(PostgresClientFuturized postgresClient, IterationJobRepository repository,
+                          Context vertxContext, Map<String, String> okapiHeaders) {
```

Comment:
Personally, I think there s no need to indent method signatures this much.

```suggestion
  public IterationJobRunner(PostgresClientFuturized postgresClient, IterationJobRepository repository,
      Context vertxContext, Map<String, String> okapiHeaders) {
```